src/item.h

```diff
@@ -704,8 +704,9 @@ class item : public JsonSerializer, public JsonDeserializer
         /**
          * Callback when a player starts wielding the item. The item is already in the weapon
          * slot and is called from there.
+         * @param mv number of moves *already* spent wielding the weapon
          */
-        void on_wield( player &p );
+        void on_wield( player &p, int mv = 0 );
```

Comment:
> The check for whether it's a worn item, in worn container or from main inventory gets a bit complicated, through.

Having those available in `on_wield` looks like it could result in spaghetti code with no clear rules what handles what.

Maybe the `wield` function should be split into "acquire from inventory" (which would do checks where is our weapon located and add own move cost) and "actually wield"?
That way we could skip the "acquirement" stage when pulling stuff out of a holster, because the holster `iuse` would provide the item and move cost.

As for messages about slow wielding: would just checking move difference be too hacky here?